File: service/src/main/java/net/huray/backend/minuting/repository/PermissionRepository.java
package net.huray.backend.minuting.repository;

import net.huray.backend.minuting.entity.Member;
import net.huray.backend.minuting.entity.Permission;
import net.huray.backend.minuting.entity.Team;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface PermissionRepository extends JpaRepository<Permission, Long> {

    List<Permission> findDistinctByMemberOrTeam(Member member, Team team);
}

File: service/src/main/kotlin/net/huray/backend/minuting/repository/PermissionRepository.kt
package net.huray.backend.minuting.repository

import net.huray.backend.minuting.entity.MemberEntity
import net.huray.backend.minuting.entity.PermissionEntity
import net.huray.backend.minuting.entity.SpaceEntity
import org.springframework.data.jpa.repository.JpaRepository

interface PermissionRepository : JpaRepository<PermissionEntity, Long> {

    fun findByMember(member: MemberEntity): List<PermissionEntity>
    fun findBySpaceAndMember(space: SpaceEntity, member: MemberEntity): PermissionEntity?

}